utils/synth_wav.sh

```diff
@@ -0,0 +1,250 @@
+n_fft=1024    # number of fft points
+n_shift=256   # number of shift points
+win_length="" # window length
+cmvn=
+
+# dictionary related
+dict=
+
+# x-vector related
+use_input_wav=false
+input_wav="input.wav"
+
+# decoding related
+synth_model=
+decode_config=
+decode_dir=decode
+griffin_lim_iters=1000
+
+# download related
+models=libritts.v1
+
+. utils/parse_options.sh || exit 1;
+
+# make shellcheck happy
+train_cmd=
+decode_cmd=
+
+. ./cmd.sh
+
+txt=$1
+download_dir=${decode_dir}/download
+
+if [ $# -gt 1 ]; then
+    echo "Usage: $0 <wav>"
+    exit 1;
+fi
+
+set -e
+set -u
+set -o pipefail
+
+case "${models}" in
+    "libritts.v1")
+        share_url="https://drive.google.com/open?id=1iAXwC0AuWusa9AcFeUVkcNLG0I-hnSr3"
+        use_input_wav=true
+        ;;
+    *)
+        echo "No such models: ${models}"
+        exit 1
+        ;;
+esac
+
+function download_models () {
+    dir=${download_dir}/${models}
+    mkdir -p ${dir}
+    if [ ! -e ${dir}/.complete ]; then
+        download_from_google_drive.sh ${share_url} ${dir} ".tar.gz"
+	touch ${dir}/.complete
+    fi
+}
+
+# Download trained models
+if [ -z "${cmvn}" ]; then
+    download_models
+    cmvn=$(find ${download_dir}/${models} -name "cmvn.ark" | head -n 1)
+fi
+if [ -z "${dict}" ]; then
+    download_models
+    dict=$(find ${download_dir}/${models} -name "*_units.txt" | head -n 1)
+fi
+if [ -z "${synth_model}" ]; then
+    download_models
+    synth_model=$(find ${download_dir}/${models} -name "model*.best" | head -n 1)
+fi
+if [ -z "${decode_config}" ]; then
+    download_models
+    decode_config=$(find ${download_dir}/${models} -name "decode*.yaml" | head -n 1)
+fi
+if [ -z "${txt}" ]; then
+    download_models
+    txt=$(find ${download_dir}/${models}/etc -name "*.txt" | head -n 1)
+fi
+
+# Check file existence
+if [ ! -f "${cmvn}" ]; then
+    echo "No such CMVN file: ${cmvn}"
+    exit 1
+fi
+if [ ! -f "${dict}" ]; then
+    echo "No such dictionary: ${dict}"
+    exit 1
+fi
+if [ ! -f "${input_wav}" ] && [ ${use_input_wav} ]; then
+    echo "No such WAV file for extracting meta information: ${input_wav}"
+    exit 1
+fi
+if [ ! -f "${synth_model}" ]; then
+    echo "No such E2E model: ${synth_model}"
+    exit 1
+fi
+if [ ! -f "${decode_config}" ]; then
+    echo "No such config file: ${decode_config}"
+    exit 1
+fi
+if [ ! -f "${txt}" ]; then
+    echo "No such txt file: ${txt}"
+    exit 1
+fi
+
+base=$(basename $txt .txt)
+decode_dir=${decode_dir}/${base}
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    echo "stage 0: Data preparation"
+
+    mkdir -p ${decode_dir}/data
+    echo "$base ${decode_dir}/data/dummy.wav" > ${decode_dir}/data/wav.scp
+    echo "X $base" > ${decode_dir}/data/spk2utt
+    echo "$base X" > ${decode_dir}/data/utt2spk
+    echo -n "$base " > ${decode_dir}/data/text
+    cat $txt >> ${decode_dir}/data/text
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    echo "stage 1: Dummy Feature Generation"
```

Comment:
I want to remove this part.
Why don't you add following lines in just before this part?
https://github.com/espnet/espnet/blob/c69b202e9cbdd0904c35ad256a44dcc177a5a318/utils/data2json.sh#L103
```bash
# modified as follows:
    if [ -z "$(find "${tmpdir}/${intype}" -name "*.scp")" ]; then
        continue
    fi
    if [ ${intype} != other ]; then 
```

Then, we can get  json like this:
```json
/home/t_hayashi/workspace5/work/espnet_dev/utils/data2json.sh decode/sample1/data decode/download/libritts.v1/data/lang_1char/train_clean_460_units.txt
{
    "utts": {
        "sample1": {
            "input": [],
            "output": [
                {
                    "name": "target1",
                    "shape": [
                        105,
                        78
                    ],
                    "text": "This example text is a how-to-write sample, so please add to and delete from it as required before using.",
                    "token": "T h i s <space> e x a m p l e <space> t e x t <space> i s <space> a <space> h o w - t o - w r i t e <space> s a m p l e , <space> s o <space> p l e a s e <space> a d d <space> t o <space> a n d <space> d e l e t e <space> f r o m <space> i t <space> a s <space> r e q u i r e d <space> b e f o r e <space> u s i n g .",
                    "tokenid": "34 50 51 61 13 47 66 43 55 58 54 47 13 62 47 66 62 13 51 61 13 43 13 50 57 65 8 62 57 8 65 60 51 62 47 13 61 43 55 58 54 47 7 13 61 57 13 58 54 47 43 61 47 13 43 46 46 13 62 57 13 43 56 46 13 46 47 54 47 62 47 13 48 60 57 55 13 51 62 13 43 61 13 60 47 59 63 51 60 47 46 13 44 47 48 57 60 47 13 63 61 51 56 49 9"
                }
            ],
            "utt2spk": "X"
        }
    }
}

```

Reply:
Thank you for your suggestion. It will make the script simple.